content/posts/about_inlining.md

```diff
@@ -28,22 +28,23 @@ Source : [inline specifier – cppreference.com][1]
 What we need to notice from that are two things:
 
 - `inline` is used to avoid function overhead, replacing the call by the implementation at compile time.
-- This keyword is only a ***hint***, not a ***command***, given to the compiler. Some functions declared inline may not really be inlined, and vice-versa (I really wish the keyword will be `maybe_inline` in the future).
+- This keyword is only a ***hint***, not a ***command***, given to the compiler. No matter how you designate a function as `inline`, it is a request that the compiler is ***allowed to ignore***: the compiler **might** inline-expand some, all, or none of the places where you call a function designated as inline (I really wish the keyword will be `maybe_inline` in the future).
 
 With that in mind, let’s take a tour of the pros and the cons of this feature.
 
 ## Pros
-First of all, even if it’s a ***hint***, the keyword **still** has a chance to influence the compiler. It may not, but it also may. For instance, we can think of the stupidest compiler that will just strictly respect the `inline` keyword, without trying to analyze the context and optimize itself. This is not against the standard (according to the C++11 standard §12.1.5) and the keyword ***is*** useful in this case.
+First of all, even if it is just a ***hint***, the keyword **still** has a chance to influence the compiler. It may not, but it also may. For instance, we can think of the stupidest compiler that will just strictly respect the `inline` keyword, without trying to analyze the context and optimize itself. This is not against the standard (according to the C++11 standard §12.1.5) and the keyword ***is*** useful in this case.
+Do not get discouraged if that seems hopelessly vague. The flexibility of the above is actually a huge advantage: it lets the compiler treat large functions differently from small ones, plus it lets the compiler generate code that is easy to debug if you select the right compiler options.
```

Comment:
[Refactor] use "which" instead of "that" in "the stupidest compiler that will" and "generate code that is"
[Refactor] after "seem" is "to V", so "seems hopelessly vague" should be "seems to be hopelessly vague"
[Suggestion] If you want to break paragraph, use an empty line. If you use single line break, in some case this method still results in 1 paragraph.